modules/core/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobStoreConfiguration.java

```diff
@@ -412,6 +423,33 @@ protected TransferManager createTransferManager() {
                                      .build();
     }
 
+    protected ObjectLockRetentionMode getRetentionMode() {
+        GetObjectLockConfigurationRequest request = new GetObjectLockConfigurationRequest().withBucketName(bucketName);
+        GetObjectLockConfigurationResult result;
+        try {
+            result = amazonS3.getObjectLockConfiguration(request);
+        } catch (AmazonServiceException e) {
+            return DEFAULT_RETENTION_MODE;
+        }
+        ObjectLockConfiguration config = result.getObjectLockConfiguration();
+        if (config == null) {
+            return DEFAULT_RETENTION_MODE;
+        }
+        ObjectLockRule rule = config.getRule();
+        if (rule == null) {
+            return DEFAULT_RETENTION_MODE;
+        }
+        DefaultRetention defaultRetention = rule.getDefaultRetention();
+        if (defaultRetention == null) {
+            return DEFAULT_RETENTION_MODE;
+        }
+        String mode = defaultRetention.getMode();
+        if (mode == null) {
+            return DEFAULT_RETENTION_MODE;
+        }
+        return ObjectLockRetentionMode.valueOf(mode);
+    }
```

Comment:
If Optional creation is not an issue, you could save some lines:
```
return Optional.ofNullable(result.getObjectLockConfiguration())
                .map(ObjectLockConfiguration:: getRule)
                .map(ObjectLockRule:: getDefaultRetention)
                .map(DefaultRetention::getMode)
                .map(ObjectLockRetentionMode::valueOf)
                .orElse(DEFAULT_RETENTION_MODE);
```

Reply:
Nice, I updated the code.